### DISCUSSION TITLE
Backward incompatible changes?

### DISCUSSION THREAD

**Discussion by mkinzie** [gollum/gollum#1763] [Q&A] [answered]
Hi - I'm a longtime admin of a Gollum wiki for my project, and I recently upgraded from version 4.1.2 to 5.2.3.  I didn't want to mess with my in-use server, so I installed it on a new VM, and copied my Git content repo there.  It worked, except all my links were broken.  After looking at all the issues and the wiki and not finding anything to say that there was a change that wasn't backward compatible, I found the command line arg --lenient-tag-lookup, and that fixed it.  

Is there any further explanation of this?  I still have a couple of links that remain broken - not sure why.  Also, the title at the top of each page has hyphens between the words.  Not important, but it looks funny.  Is there something that needs to change about the file names?

**Comment by bartkamphorst** [comment, marked as answer]
Have a look at the section [filename handling](https://github.com/gollum/gollum/wiki/5.0-release-notes#filename-handling) from the [5.0 release notes](https://github.com/gollum/gollum/wiki/5.0-release-notes). That should at least give you some idea of the changes and the reasoning behind those changes. There is also a `gollum-migrate-tags` script to help you make the transition. Hope this is of help.

**Reply by mkinzie** [reply]
Thanks.  I saw the table of contents on the Home page and mistakenly thought that was all there was in the Wiki, and that the TOC on the right was the same thing.  That helps.

**Comment by dometto** [comment]
Also FYI, Gollum follows [Semantic Versioning](https://semver.org/), so a major version update (e.g. from 4.0 -> 5.0) always indicates backwards incompatible changes. We also outlined the major changes in `5.0` in [HISTORY.md](https://github.com/gollum/gollum/blob/master/HISTORY.md#50--2020-03-17)
